Sources/Dejavu/GRDBSession.swift

```diff
@@ -630,3 +493,192 @@ final class GRDBSession: SessionInternal, @unchecked Sendable {
         }
     }
 }
+
+extension GRDBSession /* Transactions */ {
+    struct Transaction: Sendable {
+        var request: URLRequest
+        var dejavuRequest: Request
+        var instanceCount: Int
+        var response: URLResponse?
+        var data: Data?
+        var error: Error?
+    }
+}
+
+extension GRDBSession: DejavuNetworkObservationHandler {
```

Comment:
The GRDBSession for observation handler is here, but the extension for interception handler is in another file. Should both be in here? It doesn't seem like that other file needs to know about GRDB.

Reply:
Yes, they could both be in here. I actually preferred to keep them both out (leaving this file smaller), but this one needed to be in this file in order to efficiently update the private state.